src/database/schema/ChatSchema.ts

```diff
@@ -354,6 +354,29 @@ class ChatSchema implements BaseDbSchema {
     }
   };
 
+  static async deleteChat(localDb: SQLiteDatabase, messageId: string): Promise<ChatSchema> {
+    const existingChat = await ChatSchema.getByid(localDb, messageId);
+    return new ChatSchema({
+      ...existingChat,
+      type: 'deleted',
+      message: 'This message has been deleted'
+    });
+  }
+
+  static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
+    try {
+      const updateQuery = `UPDATE ${ChatSchema.getTableName()}
+        SET type = ?, message = ?
+        WHERE id = ?;`;
+
+      const updateReplacement = ['deleted', 'This message has been deleted', messageId];
+
+      await db.executeSql(updateQuery, updateReplacement);
+    } catch (e) {
+      console.log('error updating deleted chat:', e);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `deleteChat` and `updateDeletedChatType` methods provides functionality to delete a chat and update its type to 'deleted'. This is a good change as it allows for better management of chats in the database. However, I would suggest handling the error in a more robust way than just logging it to the console. Perhaps you could throw an error or return a specific result indicating failure.
```diff
+   static async deleteChat(localDb: SQLiteDatabase, messageId: string): Promise<ChatSchema> {
+     const existingChat = await ChatSchema.getByid(localDb, messageId);
+     return new ChatSchema({
+       ...existingChat,
+       type: 'deleted',
+       message: 'This message has been deleted'
+     });
+   }
+ 
+   static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
+     try {
+       const updateQuery = `UPDATE ${ChatSchema.getTableName()}
+         SET type = ?, message = ?
+         WHERE id = ?;`;
+ 
+       const updateReplacement = ['deleted', 'This message has been deleted', messageId];
+ 
+       await db.executeSql(updateQuery, updateReplacement);
+     } catch (e) {
+       console.log('error updating deleted chat:', e);
+     }
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->